include/revng/PipelineC/PipelineC.h

```diff
@@ -0,0 +1,378 @@
+#pragma once
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#ifdef __cplusplus
+#include <cstdint>
+
+#include "revng/Pipeline/Container.h"
+#include "revng/Pipeline/Kind.h"
+#include "revng/Pipeline/Runner.h"
+#include "revng/Pipeline/Step.h"
+#include "revng/Pipeline/Target.h"
+#include "revng/Pipes/PipelineManager.h"
+#else
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+#ifdef __cplusplus
+typedef revng::pipes::PipelineManager rp_runner;
+#else
+typedef struct rp_runner rp_runner;
+#endif
+
+#ifdef __cplusplus
+typedef const pipeline::Kind rp_kind;
+#else
+typedef struct rp_kind rp_kind;
+#endif
+
+#ifdef __cplusplus
+typedef pipeline::Step rp_step;
+#else
+typedef struct rp_step rp_step;
+#endif
+
+#ifdef __cplusplus
+typedef pipeline::ContainerSet::value_type rp_container_checkpoint;
+#else
+typedef struct rp_container_checkpoint rp_container_checkpoint;
+#endif
+
+#ifdef __cplusplus
+typedef const pipeline::ContainerFactorySet::value_type rp_container;
+#else
+typedef struct rp_container rp_container;
+#endif
+
+#ifdef __cplusplus
+typedef const pipeline::Target rp_target;
+#else
+typedef struct rp_target rp_target;
+#endif
+
+#ifdef __cplusplus
+typedef const pipeline::TargetsList rp_container_status;
+#else
+typedef struct rp_container_status rp_container_status;
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Every pointer returned by this library is owned by the library and must not
+ * be deleted, except for pointers returned by functions containing the word
+ * "create". Objects returned by functions named "create" are owned by the
+ * caller and must be destroyed invoking the appropriate function called
+ * rp_TYPE_destroy.
+ */
+
+/**
+ * Must be invoked before any other rp_* function is used,
+ * must be invoked exactly once.
+ * This will take care of initializing all llvm related
+ * stuff needed by the revng pipeline to operate.
+ *
+ * Do not initialize all stuff on your own.
+ */
+bool rp_init(int argc,
+             char *argv[],
+             int libraries_count,
+             const char *libraries_path[]);
+
+/**
+ * pipeline_path, flags, execution_directory cannot be null.
```

Comment:
Use Doxygen way of documenting individual arguments.

Also, describe what this does first. Afterwards you  have alternatives (`rp_runner_create_from_string` and `rp_runner_create_memory_only`) that say do the some thing as this except..., but you don't say what *this function* does.